BackEnd/src/main/kotlin/cub/trackmyoffer/FeatureProviderRouting.kt

```diff
@@ -95,7 +96,18 @@ fun Route.featureProviderRouting(httpClient: HttpClient, config: FeatureProvider
             post("/DEBUG/profile") {
                 val profileReq = call.receive<ProfileData>()
 
-                profileReq.id = 1//extractUserId(call)
+                // For DEBUG endpoints, we still need a valid user ID
+                // Get the user session and extract user info
+                val userSession: UserSession? = call.sessions.get()
+                val profileId = if (userSession != null && validateToken(httpClient, userSession)) {
+                    val userInfo = getUserInfo(httpClient, userSession)
+                    utilityDatabase.getOrCreateProfileId(userInfo.email, userInfo)
+                } else {
+                    // Fallback to 1 only if no valid session exists
+                    1
+                }
+
+                profileReq.id = profileId
```

Comment:
Thanks Junie, now please extract it in a separate method.

UPD: oh, or don't, check Mark's comment